site/src/sphinx/client-decorator.rst

```diff
@@ -67,6 +67,116 @@ If your decorator is expected to be reusable, it is recommended to define a new
     // Using a lambda expression:
     cb.decorator(delegate -> new AuditClient(delegate));
 
+The order of decoration
+-----------------------
+
+The decorators are executed in reverse order of the insertion. The following example shows which order
+the decorators are executed by printing the messages.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.WebClient;
+
+    ClientBuilder cb = Clients.builder(...);
+
+    // #2 decorator
+    cb.decorator((delegate, ctx, req) -> {
+        System.err.println("Secondly, executed.");
+        ...
+    });
+
+    // #1 decorator.
+    // No matter decorator() or option() is used, decorators are executed in reverse order of the insertion.
+    cb.option(ClientOption.DECORATION, ClientDecoration.of((delegate, ctx, req) -> {
+        System.err.println("Firstly, executed");
+        ...
+    });
+
+    WebClient myClient = cb.build(WebClient.class);
+
+The following diagram describes how an HTTP request and HTTP response are gone through decorators:
+
+.. uml::
+
+    @startditaa
+    +-----------+  req  +-----------+  req  +-----------+  req  +------------------+    req    +--------------+
+    |           |------>|           |------>|           |------>|                  |---------->|              |
+    | WebClient |       | #1        |       | #2        |       | Armeria          |           |    Server    |
+    |           |  res  | decorator |  res  | decorator |  res  | Networking Layer |    res    |              |
+    |           |<------|           |<------|           |<------|                  |<----------|              |
+    +-----------+       +-----------+       +-----------+       +------------------+           +--------------+
+    @endditaa
+
+If the client is a Thrift client and RPC decorators are inserted, HTTP decorators and RPC decorators are
```

Comment:
I haven't really used RPC decorators before since don't use thrift. I'm surprised by this behavior, and understand it but it's not very intuitive to have implicit grouping.

Would it be worth changing the API a little to make the grouping explicit?

```java
cb.decorators().add()...

cb.rpcDecorators().add()...
```

Reply:
That's a good suggestion.
The example code that I picked up was really ugly and I think there's a still chance that a user does it in the same way:
```java
cb.decorators().add(d1);
cb.rpcDecorators().add(d2);
cb.decorators().add(d3);
```
And mostly, a user would have separate methods to set decorators if it gets complicated, so I'm not sure it's worth changing. 🤔 (But it looks good to me.)